cql/ManageRareAbnormality.cql

```diff
@@ -627,6 +630,70 @@ define HistologyInterpretedAsCin3:
         aC in Dash."Histologic CIN3"
   )
 
+define AnyHistologyInterpretedAsCin3:
+  AnyTrue(
+    (Collate.SortedBiopsyReports.allConclusions) aC
+      return
+        aC in Dash."Histologic CIN3"
+  )
+
+define AnyHistologyInterpretedAsCin2:
+  AnyTrue(
+    (Collate.SortedBiopsyReports.allConclusions) aC
+      return
+        aC ~ Dash."CIN2"
+  )
+
+define MostSevereHistologyInterpretedAsCin2InPast3Years:
+  AnyTrue(
+    (Collate.SortedBiopsyReports.allConclusions) aC
+      return
+        aC ~ Dash."CIN2"
+  )
+
+define MostRecentCin2BiopsyInPast3Years:
+  Coalesce(
+    (Collate.SortedBiopsyReports) S
+    where (S.date >= Now() - 3 years) and
+    AnyTrue(
+    (S.allConclusions) aC
+      return 
+        aC ~ Dash."CIN2"
+    )
```

Comment:
It looks like the AnyHistologyInterpretedAsCin2 expression you defined above already does this. Consider using that expression in these follow-on expressions, so that you don't have to keep redefining the same logic.

Reply:
This is iterating over a sorted array, and stopping at the first match, so it is returning the most recent biopsy. AnyHistologyInterpretedAsCin2 will look through all results for CIN2.

Reply:
I think you need to take the order of operations here. `Coalesce` is called on the outside of this expression and is the last thing to evaluate. This is not returning the first entry of `SortedBiopsyReports`. This is returning the first entry of a subset of that array, after you have:
1. Filtered by date
2. Filtered by only biopsy reports that have a result of CIN2. 

Coalesce is called after that sorting has been done, so you are returning the most recent of the CIN2 biopsy results in the past 3 years, even if it was not the most recent overall biopsy report.

Reply:
Discussed this with @mrnosal and we decided on the labelling of "LastKnownX" here to avoid confusion with other expressions in our CQL that are titled "MostRecentX". As for the initial comment about repeating the filtering logic of the biopsies several times in this section, I have opened the following Jira ticket to address this in a follow-on refactor:

**CQL Refactor: Organize and Combine Biopsy Lookbacks**: [CCSMCDS-81](https://jira.mitre.org/browse/CCSMCDS-81)